storage-proofs/src/batchpost.rs

```diff
@@ -246,21 +246,29 @@ mod tests {
 
         let priv_inputs = PrivateInputs::<H>::new(data.as_slice(), &tree);
 
-        let proof = BatchPoST::<H>::prove(&pub_params, &pub_inputs, &priv_inputs).unwrap();
+        let proof =
+            BatchPoST::<H>::prove(&pub_params, &pub_inputs, &priv_inputs).unwrap_or_else(|e| {
+                panic!("Failed to create proof: {}", e);
+            });
```

Comment:
The idea was to use [`.expect`](https://doc.rust-lang.org/std/result/enum.Result.html#method.expect) for these. 

that lets you write this as 

```rust
do_fail_thing().expect("do fail thing failed");
```

Reply:
Isn't the information in the error worth it?

Reply:
using `expect` will give you that information, take a look at this playground example: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=8e81bc6b0ae9dee0102e196dc3e83967

Reply:
`expect` prints the `Debug` info, whereas `panic!("{}", e)` prints the `Display`, I used the later because `error::Error` has the nice `Fail` display attributes on its variants (e.g. https://github.com/filecoin-project/rust-fil-proofs/blob/master/storage-proofs/src/error.rs#L8) and imo gives better context.

I'm happy to use either `expect` or `panic`.

Reply:
Good point, I think both options serve the purpose of #390 quite nicely so I'd favor keeping the PR as is and avoid duplicating work.

Reply:
@dignifiedquire are you in favor of using `expect` everywhere?

Reply:
I think expect is much more idiomatic rust, and much easier to read, so lets use that. If we need more detailed views on panic, we should try to improve our Error type.